### ISSUE TITLE
Add in documentation that Node.js global vars may be kept between deployments

### ISSUE THREAD

**Issue by paulbatum** [Azure/Azure-Functions#368]
_From @yvele on February 24, 2017 20:20_

Specially usefull information when using Webpack to [drastically reduce Azure Function cold start](https://github.com/Azure/azure-webjobs-sdk-script/issues/298#issuecomment-281779696).

I understand the global vars sharing when I tried to use `babel-polyfill` and got this error:

> Error: only one instance of babel-polyfill is allowed

[Babel polyfill](https://babeljs.io/docs/usage/polyfill/) is using global vars and although I was importing it only once (in my `webpack.config`), I constantly get this error, and I finally had to make a little wrapper around my Azure Function handle to make sure `babel-polyfill` is not already loaded in global vars before requiring it.

I suggest adding a global vars warning in the documentation (especially with all Node.js developers going to make a webpack bundle to [reduce Azure Function cold start](https://github.com/Azure/azure-webjobs-sdk-script/issues/298#issuecomment-281779696).

This issue is kind of related to https://github.com/Azure/azure-webjobs-sdk-script/issues/298#issuecomment-281779696 and attempts to make Azure Function work with Webpack.

_Copied from original issue: Azure/azure-webjobs-sdk-script#1234_

**Comment by paulbatum** [comment]
_From @lindydonna on March 14, 2017 0:14_

@christopheranderson @yochay Shouldn't this be moved to the Functions repo, since it's about documentation?

**Comment by paulbatum** [comment]
_From @christopheranderson on April 3, 2017 17:17_

@lindydonna - yes this should be moved

**Comment by paulbatum** [comment]
_From @lindydonna on April 3, 2017 19:35_

@christopheranderson Could you file a new issue in the docs repo to track this?  You have more context on the doc requirement.

**Comment by nevercast** [comment]
@yvele What was your workaround for this?

**Comment by ColbyTresness** [comment]
@mhoeger thoughts here?

**Comment by mhoeger** [comment]
I'm a bit confused, I'm suspicious that this is a functions issue? https://github.com/babel/babel-loader/issues/401

I think it would be good though to add that global state is global for the _function app_ (not per function), but there are no guarantees about preserved state of a global var (alluded to [here](https://docs.microsoft.com/en-us/azure/azure-functions/functions-reference-node#using-entrypoint))

**Comment by ColbyTresness** [comment]
I'm comfortable closing then. Please repoen if the doc Marie referenced is insufficient.
